### ISSUE TITLE
Binary DL Export is somehow broken

### ISSUE THREAD

**Issue by aGitrHUbUser** [Fast-64/fast64#458]
it is exported sucessfully but when i tried to change the geo dl to load the exported one, it crashed

**Comment by Lilaa3** [comment]
Does your mesh happen to have multiple materials?

**Comment by Lilaa3** [comment]
OK so as one does I learned how to use the exporter myself and i think i know what is causing your confusion, yes the displaylist exporter seems to be outdated for binary (ill pr to fix that) but this is probably because you're trying to use the geolayout address (like the name would suggest) instead of the address in the command itself, so say you want to update the sand dl, the load from dl command is at 0x2ABD88, but the address that points to the dl is at 0x2ABD8C, does that make sense?

**Comment by aGitrHUbUser** [comment]
> Does your mesh happen to have multiple materials?

Only one material

**Comment by Lilaa3** [comment]
> > Does your mesh happen to have multiple materials?
> 
> Only one material

Could you respond to my other comment?

**Comment by aGitrHUbUser** [comment]
> OK so as one does I learned how to use the exporter myself and i think i know what is causing your confusion, yes the displaylist exporter seems to be outdated for binary (ill pr to fix that) but this is probably because you're trying to use the geolayout address (like the name would suggest) instead of the address in the command itself, so say you want to update the sand dl, the load from dl command is at 0x2ABD88, but the address that points to the dl is at 0x2ABD8C, does that make sense?

If you're talking about LOAD_MODEL_FROM_DL or LOAD_MODEL_FROM_GEO function, not really, i exported it without overwrite geolayout or dl pointers, only export it and change bully's GEO horn to load another display list

**Comment by Lilaa3** [comment]
yes, the address you used is probably not the start for the main display list, if you know the address of the display list commands's display list you could tell fast64 to override that like you would for load model from dl

**Comment by aGitrHUbUser** [comment]
> yes, the address you used is probably not the start for the main display list, if you know the address of the display list command's display list you could tell fast64 to override that like you would for load model from dl

i did looked in the hxd and it seem to have 3 fast3d commands that linked to 3 displaylists, i load one of them in a door geo and it came out like this
![image](https://github.com/user-attachments/assets/5ffe786c-2ad4-4c59-8dd5-f4e459b82bbf)

**Comment by Lilaa3** [comment]
material, geometry and material revert specifically! 
does the door render correctly if you use the main displaylist with all 3? 
also are you using the correct level option? that's used to convert the virtual address to the correct segmented address

**Comment by Lilaa3** [comment]
0x4035378 should be correct, do you have a blend by the way?

**Comment by aGitrHUbUser** [comment]
> 0x4035378 should be correct, do you have a blend you can share by the way?

didn't have a blend plus uh it's a cube
![image](https://github.com/user-attachments/assets/8cb64a2c-b771-4ec4-9b43-a03428283a1c)

**Comment by jesusyoshi54** [comment]
Was this fixed by PR #459 
As far as I'm aware, after that commit binary exporting should have worked. I will close this if there is no reply in a week or whenever I remember.
